Library/Homebrew/cask/lib/hbc/utils/quarantine.swift

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/swift
+
+import Foundation
+import CoreServices
+
+let QUARANTINE_SUCCESS: Int = 0
+let QUARANTINE_FAILURE: Int = 255
+
+if (CommandLine.arguments.count < 3) {
+  print("Insufficient parameters: this script needs file_path and data_url")
+  exit(Int32(QUARANTINE_FAILURE))
+}
+
+let filepath = CommandLine.arguments[1]
+let agent: String = "Homebrew-Cask"
+let bundle: String = "sh.brew.cask"
```

Comment:
Could this be a path instead? I think I remember macOS using the path to the executable instead of a bundle ID if it's not an app bundle.

Reply:
This one is inferred from [the parent process's bundle identifier](https://github.com/phracker/MacOSX-SDKs/blob/9fc3ed0ad0345950ac25c28695b0427846eea966/MacOSX10.13.sdk/System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/Headers/LSQuarantine.h#L50-L54); since this is Swift's interpreter, it'd be set to `nil`.

Reply:
In Apple's docs [here](https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFBundles/BundleTypes/BundleTypes.html#//apple_ref/doc/uid/10000123i-CH101-SW1) a bundle identifier is a reverse-DNS [Uniform Type Identifier](https://developer.apple.com/library/archive/documentation/Miscellaneous/Reference/UTIRef/Articles/System-DeclaredUniformTypeIdentifiers.html); I don't know if Gatekeeper enforces this, though.

Reply:
I think the `TCC.db` uses paths for executables and the bundle ID for apps, and this seems similar to that.

Reply:
I've removed this altogether in an upcoming commit (it's not needed at all).